shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -97,6 +97,34 @@ - (NSString*)accessibilityValue {
   }
 }
 
+- (BOOL)accessibilityActivate {
+  return [_semanticsObject accessibilityActivate];
+}
+
+- (void)accessibilityIncrement {
+  [_semanticsObject accessibilityIncrement];
+}
+
+- (void)accessibilityDecrement {
+  [_semanticsObject accessibilityDecrement];
+}
+
+- (BOOL)accessibilityScroll:(UIAccessibilityScrollDirection)direction {
+  return [_semanticsObject accessibilityScroll:direction];
+}
+
+- (BOOL)accessibilityPerformEscape {
+  return [_semanticsObject accessibilityPerformEscape];
+}
+
+- (void)accessibilityElementDidBecomeFocused {
+  [_semanticsObject accessibilityElementDidBecomeFocused];
+}
+
+- (void)accessibilityElementDidLoseFocus {
```

Comment:
Did you actually have to do this?  I would have expected that the combination of`methodSignatureForSelector:` and `forwardInvocation:` should have given you the same behavior.  Any of these methods that just forward to `_semanticsObject` should be handled with that, unless it isn't functioning properly.

That code is intended to say, "If I handle the method, execute it.  If I don't, forward it to `_semanticsObject`".

Reply:
In practice it doesn't work without this change :)

Does that forwarding logic work as described if the base class has an implementation for accessibilityElementDidLoseFocus?

Reply:
(The test I am adding is also failing if I remove these manual forwards)

Reply:
Hmm, yea the existence of the other methods that manually forward the methods seems to suggest they are needed.  I'm not sure why yet.  Give me a sec to investigate.  I think we should add a comment or remove some code depending on what I find.

Reply:
Ahh, ok I remember now.  This guy inherits from the UISwitch, so if the UISwitch handles it, it forwards it to him, otherwise if it is manually declared it forwards to the semantics object.

Reply:
The problem is that the required information is in the header file.  I don't have a good idea about what a comment should say.  This LGTM, some comment in the .mm file would be helpful.

Reply:
Added a comment.